### PR TITLE
Merge nodejs/master

### PR DESCRIPTION
**Pull request by kunalspathak**
Merge f84666f923 as of 2017-05-28.

Commit to review -
65af6d5 - Implemented `napi_get_version` api for JSRT.

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

n-api, core

### CODE REVIEW

File: src/node_api.cc [hunk 1]
```diff
@@ -18,8 +18,6 @@
 #include "uv.h"
 #include "node_api.h"
 
-#define NAPI_VERSION  1
```
Comment (kunalspathak) [inline src/node_api.cc]: I will submit this change upstream. I am checking with Michael [here](https://github.com/nodejs/node/pull/13207/files#r119427792) if he has strong reason for doing it this way.

**Comment by kunalspathak** [comment]
CI - https://ci.nodejs.org/job/chakracore-test-pull-request/60/

**Comment by kfarnung** [comment]
Looks like the linter is failing in CI.

There's also a failure that we seem to be seeing on Ubuntu 14.04: `sequential/test-benchmark-net`.

We're also still seeing the following error class on macOS and Ubuntu 16.04:
```
# Clean up any leftover processes, error if found.
ps awwx | grep Release/node | grep -v grep | cat
32726 ?        RLl    1:40 /home/iojs/build/workspace/chakracore-test-linux/nodes/ubuntu1604-64/out/Release/node /home/iojs/build/workspace/chakracore-test-linux/nodes/ubuntu1604-64/test/parallel/test-vm-sigint.js child runInContext 0
32733 ?        RLl    1:40 /home/iojs/build/workspace/chakracore-test-linux/nodes/ubuntu1604-64/out/Release/node /home/iojs/build/workspace/chakracore-test-linux/nodes/ubuntu1604-64/test/parallel/test-vm-sigint.js child runInContext 1
32736 ?        RLl    1:40 /home/iojs/build/workspace/chakracore-test-linux/nodes/ubuntu1604-64/out/Release/node /home/iojs/build/workspace/chakracore-test-linux/nodes/ubuntu1604-64/test/parallel/test-vm-sigint.js child runInContext 2
make[1]: *** [Makefile:358: test-ci] Error 1
make: *** [Makefile:524: run-ci] Error 2
Build step 'Execute shell' marked build as failure
```

@kunalspathak I think the latter issues are existing, but can you investigate the linker issue?

**Comment by kunalspathak** [comment]
Yes, i have already fixed the linter issue. Let me update the PR.

**Review by kfarnung — APPROVED** [review]
LGTM 🚢

**Review by digitalinfinity — APPROVED** [review]

**Pull request merged**